src/stateless/slider.vue

```diff
@@ -0,0 +1,545 @@
+
+export default {
+    components: {
+        inputElement: Input,
+    },
+    props: {
+        min: { type: Number, required: true },
+        max: { type: Number, required: true },
+        limit: { type: Number, required: false },
+        minLabel: { type: String, required: false },
+        maxLabel: { type: String, required: false },
+        label: { type: String },
+        step: { type: Number },
+        value: { type: Number },
+        unit: { type: String, required: false },
+        disabled: { type: Boolean, default: false },
+        disabledText: { type: String, required: false },
+        theme: { type: String, default: 'dark' },
+        size: { type: String, default: 'normal' },
+        alignment: { type: String, default: 'left' },
+        isValid: { type: Function, required: false },
+    },
+    data () {
+        return {
+            isDomReady: false,
+            isChanged: false,
+            isDragging: false,
+        }
+    },
+    computed: {
+        limitValue () {
+            return this.limit || this.max
+        },
+        minLabelValue () {
+            return this.minLabel || this.min.toString()
+        },
+        maxLabelValue () {
+            return this.maxLabel || this.limitValue.toString()
+        },
+        index () {
+            return (this.value - this.min) / this.step
+        },
+        stepsCount () {
+            return (this.limitValue - this.min) / this.step
+        },
+        stepPercentage () {
+            return 1 / this.stepsCount
+        },
+        percentage () {
+            return (this.value - this.min) / (this.limitValue - this.min)
+        },
+        position () {
+            return this.isDomReady ? Math.max(0, Math.min(this.index, this.stepsCount)) * this.stepPercentage * this.sliderWidth : 0
+        },
+        sliderWidth () {
+            return this.isDomReady ? this.$refs.bar.clientWidth : 0
+        },
+        sliderOffset () {
+            return this.isDomReady ? this.$refs.bar.getBoundingClientRect().x : 0
+        },
+        decimalPlacesCount () {
+            let decimals = this.step.toString().split('.')[1]
+            return decimals ? decimals.length : 0
+        },
+        inputType () {
+            return this.decimalPlacesCount === 0 ? 'number' : 'float'
+        },
+    },
+    created () {
+        if (this.decimalPlacesCount > 1)
+            throw new Error('Max one decimal place is supported.')
+
+        let isStepValid = (this.max - this.min) / this.step % 1 === 0
+        if (!isStepValid)
+            throw new Error('Difference between max and min is not divisible by step.')
+
+        let isValueValid = ((this.value - this.min) % this.step).toFixed(0) === '0'
+        if (!isValueValid)
+            throw new Error('Value is not a valid step.')
+
+        if (this.min < 0 || this.min > 998)
+            throw new Error('Min must be between 0 and 998.')
+
+        if (this.max > 999 || this.max <= this.min)
```

Comment:
Why do we require that a value is between 0 and 999? Why can't `min` and `max` be any number, with a single requirement that `max - min >= step`?

Reply:
Because UI would break if we did.